src/Microsoft.AspNetCore.OData.Authorization/ODataAuthorizationHandler.cs

```diff
@@ -0,0 +1,54 @@
+﻿// Copyright (c) Microsoft Corporation.  All rights reserved.
+// Licensed under the MIT License.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Authorization;
+
+namespace Microsoft.AspNetCore.OData.Authorization
+{
+    /// <summary>
+    /// Decides whether an OData request should be authorized or denied.
+    /// </summary>
+    internal class ODataAuthorizationHandler : AuthorizationHandler<ODataAuthorizationScopesRequirement>
+    {
+        private Func<ScopeFinderContext, Task<IEnumerable<string>>> _scopesFinder;
+
+        /// <summary>
+        /// Creates an instance of <see cref="ODataAuthorizationHandler"/>.
+        /// </summary>
+        /// <param name="scopesFinder">User-define function used to retrieve the current user's scopes from the authorization context</param>
+        public ODataAuthorizationHandler(Func<ScopeFinderContext, Task<IEnumerable<string>>> scopesFinder = null) : base()
+        {
+            this._scopesFinder = scopesFinder;
```

Comment:
```suggestion
            _scopesFinder = scopesFinder;
```
Drop the use of `this` across the files